vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1320,6 +1327,46 @@ def execute_model(
                                             all_gather_group=get_tp_group())
             logits = None
         else:
+            if self.input_batch.pooling_params:
```

Comment:
Can we move this implementation into a separate method? I feel like the `execute_model` method is too large.